contracts/Marketplace.sol

```diff
@@ -408,6 +408,14 @@ contract Marketplace is Proofs, StateRetrieval, Endian {
     _;
   }
 
+  modifier requestNotFinished(RequestId requestId) {
```

Comment:
Soooo, I was checking something around the `fillSlot` and actually I noticed that there is the line `require(slotState(slotId) == SlotState.Free, "Slot is not free");` which inside actually checks for `requestState()` and if the request is cancelled/failed/finished, then it will returns something else then `SlotState.Free`, soooo actually this issue is not relevant? 😅
https://github.com/codex-storage/codex-contracts-eth/issues/157

Reply:
Can you also please double-check this @0x-r4bbit if I see this correctly? 

https://github.com/codex-storage/codex-contracts-eth/blob/8bbb4257f7b42bf257303d4f504674adbebdefee/contracts/Marketplace.sol#L139

If so, let's then remove this modifier? I guess I would like to find maybe some better way how to communicate this in the code as it is not clear, but that is refactor for some other day.

Reply:
Oh well...

So I've removed the modifier and ran the prover on `allowedRequestStateChanges` rule (https://prover.certora.com/output/6199/c0f500e68f0441a7b7f371aa6e9cba12?anonymousKey=2014a4af0ac07d3847d0ae7b200ef07510dab545) and `functionsCausingRequestStateChanges` rule (https://prover.certora.com/output/6199/f19c47de699e4d9694f108b7693e75c5?anonymousKey=b5dccf783024345142761adb6205426ba6cfe14f) and they are both passing.

I can't remember what scenario caused a counter example where the modifier was needed.
But it looks like it no longer is.

Thanks once again for your using your eagle eyes!

Reply:
No worries, I also originally thought that the bug is valid ;-)

Reply:
@AuHau Here you go: This is now failing with all rules enabled: https://prover.certora.com/output/532476/1955a12c799c4c749d4e7f2f94549607?anonymousKey=82e229207950662a9c624d7b0864bf40b1edee4e

Reply:
Ah, that's the issue again with the prover runnign `fillSlot()` with `requestId = 0` which will always give you a `Slot.Free` slot, even though the `RequestState` is `Finished`.

Gonna fix this.

Reply:
Alright, here's a prover run with a fix: https://prover.certora.com/output/6199/0ea1d80bc3ad42438c9303d2b3604b4f?anonymousKey=5faeef8e81a4b5ee0d30f77b5ee984f1c5d26410

Here an isolated failing one for comparison:
https://prover.certora.com/output/6199/81939b2b12d74a5cae5e84ceadb901c0?anonymousKey=a4ad6268598a1077ecfce75493b0c0f9bc3b17a0

Reply:
Updated the PR with the fixes.